service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/WorkspaceCloneUtils.java

```diff
@@ -401,12 +428,16 @@ protected static List<ResourceLineageEntry> buildDestinationResourceLineage(
   }
 
   private static Map<String, String> maybeClearSomeResourcePropertiesBeforeCloning(
-      WsmResource sourceResource, UUID destinationWorkspaceId) {
+      WsmResource sourceResource, UUID destinationWorkspaceId, @Nullable UUID destinationFolderId) {
```

Comment:
It's confusing that we're setting a property in a method called `maybeClearSomeResourcePropertiesBeforeCloning()`. Suggestion to make code more readable:

```
  private static Map<String, String> maybeAdjustResourceProperties(
      WsmResource sourceResource, UUID destinationWorkspaceId, @Nullable UUID destinationFolderId) {
    Map<String, String> destinationResourceProperties = sourceResource.getProperties();
    if (!destinationWorkspaceId.equals(sourceResource.getWorkspaceId())) {
      if (destinationFolderId != null) {
        // We're cloning a workspace. Set folder ID to corresponding folder in destination workspace.
        destinationResourceProperties.put(FOLDER_ID_KEY, destinationFolderId.toString());
      } else {
        // We're cloning an individual resource. If folder ID property exists, clear it, since folder doesn't exist in destination workspace.
              destinationResourceProperties =
          clearSomePropertiesForResourceCloningToDifferentWorkspace(destinationResourceProperties);
      }
    HashMap<String, String> result = new HashMap<>(destinationResourceProperties);
    return ImmutableMap.copyOf(result);
  }
```